src/lua/api-gateway/tracking/factory.lua

```diff
@@ -37,9 +37,33 @@ local cjson = require "cjson"
 --   POST /tracking
 local function _API_POST_Handler()
     local trackingManager = ngx.apiGateway.tracking.manager
-    ngx.req.read_body()
     if ( ngx.req.get_method() == "POST" ) then
        local json_string = ngx.req.get_body_data()
```

Comment:
one thing here get_body_data should always be called after read_body ---> from official docs

> If the request body has not been read yet, call ngx.req.read_body first (or turned on lua_need_request_body to force this module to read the request body. This is not recommended however).

Reply:
And checked in confs - nobody else reads the body - only place where this is called is 

`location /tracking {
    limit_except POST {
        deny  all;
    }
    content_by_lua 'ngx.apiGateway.tracking.POST_HANDLER()';
}`

Reply:
I'll update the code.